bpf/lib/nodeport.h

```diff
@@ -137,6 +137,25 @@ static __always_inline __u64 ctx_adjust_room_dsr_flags(void)
 #endif
 }
 
+static __always_inline bool dsr_fail_needs_reply(int code __maybe_unused)
+{
+#ifdef ENABLE_DSR_ICMP_ERRORS
+	if (code == DROP_FRAG_NEEDED)
+		return true;
+#endif
+	return false;
+}
+
+static __always_inline bool dsr_is_too_big(struct __ctx_buff *ctx __maybe_unused,
+					   __u16 expanded_tot_len __maybe_unused)
+{
+#ifdef ENABLE_DSR_ICMP_ERRORS
+	if (expanded_tot_len > MTU)
```

Comment:
The MTU value might be not reliable (see https://github.com/cilium/cilium/pull/14863). Maybe makes sense to add a new macro `SELF_MTU` per each device?